### PR TITLE
chore(ci): move to gha

### PR DESCRIPTION
**Pull request by frezbo**
Move to using GHA.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          make
+      - name: Login to registry
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3
+        with:
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+      - name: Push to registry
+        if: github.event_name != 'pull_request'
```
Comment (smira) [inline .github/workflows/ci.yaml:67]: we had push only on `tag` event

Comment (frezbo) [reply .github/workflows/ci.yaml:67]: ah yes

Comment (frezbo) [reply .github/workflows/ci.yaml]: fixed

**Review by smira — APPROVED** [review]

**Pull request merged**